app/lib/linters/crunch_api.rb

```diff
@@ -0,0 +1,97 @@
+require 'rest-client'
+require 'base64'
+module Linters
+
+  # A class called Crunch
+  class CrunchApi
+
+    attr_accessor :base_url
+
+    def initialize()
+      @base_url = "https://platform.42crunch.com"
+    end
+
+    # Lints an unploaded file with 42crunch.
+    #
+    # @param file [File] the file to be linted
+    # @return [String] the raw JSON response
+    def lint_to_json(file)
+
+      # Initialize local variables
+      create_response = String.new
+      created_api_id = String.new
+      report_response = String.new
+      report_data = String.new
+      decoded_report = String.new
+
+      # Post the file to 42Crunch according to their API v1 "Create an API (from file)" spec
+      # https://www.postman.com/get-42crunch/workspace/42crunch-api/request/13761657-335bcb36-c68f-43a1-823d-741b62b55bc6
+      begin
+        create_response += RestClient.post( @base_url + "/api/v1/apis",
+        { # Body of the request
+          cid: ENV['COLLECTION_ID'], # Collection id, returned by "Create a collection"
+          name: File.basename(file.path, ".*"), # API Display Name
+          yaml: false, # Set to true if the specification file was converted to JSON from YAML
+          specfile: file # Raw OAS file in JSON format - YAML is not supported
+        },
+        { # Request headers
+        'X-API-KEY': ENV['CRUNCH_API_KEY']
+        })
+      rescue RestClient::MovedPermanently,
+             RestClient::Found,
+             RestClient::TemporaryRedirect => err
+        err.response.follow_redirection
+      rescue RestClient::Unauthorized, RestClient::Forbidden => err
+        raise 'Access denied by 42Crunch API'
+      rescue RestClient::RequestFailed => err
+        raise '42Crunch Create Request Failed'
+      end
```

Comment:
These `rescue` clauses make debugging problems harder.

The act of rescuing and then raising a new error effectively erases the information gathered in the original error. For example, the backtrace will not point to the root cause of the problem, but rather to this line in the code.